### PR TITLE
ODC-7588: Automatically select new Deployments or Knative Services in Topology after creating them with the Import from Git or Import image container flow

### PR DESCRIPTION
**Pull request by logonoff**
From Jira issue https://issues.redhat.com/browse/ODC-7588

**Fixes**: 

**Analysis / Root cause**: 

n/a

**Solution Description**: 

Adds option of adding search params to `handleRedirect`, then passes the required `selectId` upon creation of a service by filtering only deployment resources.

Needed to make a `addSearchParamsToRelativeURL` helper to include this feature, as the URL returned by `importRedirectURL` has no guarantee of not already including search params or url fragments, and naive string concatenation would result in bugs in that case. I applied [ITEF RFC 2257](https://datatracker.ietf.org/doc/html/rfc2557#section-5) spec to implement the helper that adds search params to the URL

I also updated the jsdoc for `importRedirectURL` to explicitly indicate that it returns a relative URL as a postcondition. This was already assumed by the `handleRedirect` helper, but was not explicitly written (i.e., if an absolute url were returned, it would not redirect the user to the correct url).   

I also had to modify the topology page to ignore the last opened sidebar in favor of the new id I'm giving it, which may have consequences I'm not aware of. Testing indicates that the page just saves the new ID as the new previously viewed deployment

**Screen shots / Gifs for design review**: 

https://github.com/openshift/console/assets/18614559/7be4226e-76f2-49bb-bdce-d830cec6ec2d

This screen recording was taken on a new installation of Chromium (OzonePlatform set to Wayland) 125.0.6422.141 on Fedora CSB 40 with no saved cookies or site data. No operators were installed on OpenShift

**Unit test coverage report**: 

Not changed.

**Test setup:**

No additional setup is required aside from already having an OpenShift cluster :p

Here are the URLs/container image ids I used for easy copy/pasting:
- https://github.com/sclorg/nodejs-ex.git
- https://github.com/sclorg/golang-ex.git
- https://github.com/sclorg/ruby-ex.git
- openshift/hello-openshift
- [server.jar](https://piston-data.mojang.com/v1/objects/145ff0858209bcfc164859ba735d4199aafa1eea/server.jar) (it can be any jar) 

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge - not needed as edge is now chromium based

### CODE REVIEW

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts [hunk 1]
```diff
@@ -24,7 +24,10 @@ import {
   k8sUpdate,
   K8sVerb,
 } from '@console/internal/module/k8s';
-import { ServiceModel as KnServiceModel } from '@console/knative-plugin';
+import {
+  ServiceModel as KnServiceModel,
+  ServiceModel as knSvcModel,
```
Comment (vikram-raj) [inline frontend/packages/dev-console/src/components/import/import-submit-utils.ts:29]: You importing the same model twice. Remove the duplicates.
```suggestion
  ServiceModel as KnServiceModel,
```

**Review by vikram-raj — COMMENTED** [review]
one nit suggested. Can you please add e2e tests.

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts [hunk 2]
```diff
@@ -852,14 +854,40 @@ export const createOrUpdateResources = async (
   return responses;
 };
 
+export const filterDeployedResources = (resources: K8sResourceKind[]) =>
+  resources.filter(
+    (resource) =>
+      resource.kind === DeploymentModel.kind ||
+      resource.kind === DeploymentConfigModel.kind ||
+      (resource.kind === knSvcModel.kind &&
+        resource.apiVersion === `${knSvcModel.apiGroup}/${knSvcModel.apiVersion}`),
```
Comment (vikram-raj) [inline frontend/packages/dev-console/src/components/import/import-submit-utils.ts:863]: Need to update the code accordingly as you removed the `knSvcModel` import.
```suggestion
      (resource.kind === KnServiceModel.kind &&
        resource.apiVersion === `${KnServiceModel.apiGroup}/${KnServiceModel.apiVersion}`),
```

**Comment by vikram-raj** [comment]
Propagate from the epic

/label px-approved
/label docs-approved

**Review by vikram-raj — COMMENTED** [review]
/lgtm
/approve

File: frontend/packages/topology/integration-tests/support/pages/topology/topology-page.ts [hunk 3]
```diff
@@ -16,6 +16,11 @@ import { gitPage } from '@console/dev-console/integration-tests/support/pages/ad
 import { topologyHelper } from './topology-helper-page';
 
 export const topologyPage = {
+  verifyOrOpenSidebar: (nodeName: string) => {
+    if (!cy.get(topologyPO.sidePane.dialog)) {
+      topologyPage.componentNode(nodeName).click({ force: true });
+    }
```
Comment (sanketpathak) [inline frontend/packages/topology/integration-tests/support/pages/topology/topology-page.ts:22]:
```suggestion
    cy.get('body').then(($body) => {
      if ($body.findtopologyPO.sidePane.dialog).length !== 0) {
        topologyPage.componentNode(nodeName).click({ force: true });
      } else {
        cy.log(`Sidebar is already open`);
      }
    });
```

**Comment by sanketpathak** [comment]
@logonoff The import from git in the serverless function is not working

https://github.com/openshift/console/assets/10252230/6bceb6c7-0227-4d03-87f5-65aa1ebdbb20

**Comment by logonoff** [comment]
> @logonoff The import from git in the serverless function is not working
> Screen.Recording.2024-06-21.at.6.48.17.PM.mov

Resolved

**Comment by sanketpathak** [comment]
Verified on a cluster with admin and non-admin user


https://github.com/openshift/console/assets/10252230/ab923f68-6881-43db-a3f1-d1d12d669510


/label qe-approved

**Pull request merged**